src/app/api/firebaseAuth.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  getAuth,
+  createUserWithEmailAndPassword,
+  UserCredential,
+  AuthError,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import auth from "../lib/firebase/firebaseConfig";
+
+type userAuthResponse = {
+  success: boolean;
+  userId: number;
+};
+
+export const signUpUser = (
```

Comment:
instead of using .then() for these functions, let's change them to use async/await since .then() doesn't pause the function execution, which is causing both of these functions to return a userAuthResponse with success set to false even though the request works properly